### ISSUE TITLE
Player lost ALL not actively used claim blocks

### ISSUE THREAD

**Issue by sennby** [GriefPrevention/GriefPrevention#2257]
### Observed Behavior

all claim blocks that are not claimed actively on land, were lost.
Player now has 0 available claim blocks, 0 bonus as well.

### Expected Behavior

the player shouldnt lose blocks.

### Reproduction steps

couldnt tell you.

### Server version

```shell
[21:10:54 INFO]: Checking version, please wait...
[21:10:54 INFO]: This server is running UniverseSpigot version git-UniverseSpigot-"29224da" (MC: 1.20.4) (Implementing API version 1.20.4-R0.1-SNAPSHOT) (Git: 29224da on ver/1.20.4)
Universe-git-UniverseSpigot-"29224da" (MC: 1.20.4)
```

### GriefPrevention version

```shell
version griefprevention
[21:11:14 INFO]: GriefPrevention version 16.18.2
```

### Configuration

```yaml
# Default values are perfect for most servers.  If you want to customize and have a question, look for the answer here first: http://dev.bukkit.org/bukkit-plugins/grief-prevention/pages/setup-and-configuration/

GriefPrevention:
  SeaLevelOverrides:
    world: -1
    world_nether: -1
    world_the_end: -1
    clanwars: -1
    QueenBeeWorld: -1
    arena: -1
    dungeon: -1
    Boss: -1
    spawn: -1
    MobArena: -1
    resource_world: -1
  Claims:
    Mode:
      world: Survival
      QueenBeeWorld: Disabled
      arena: Disabled
      clanwars: Disabled
      resource_world: Disabled
      Boss: Disabled
      spawn: Disabled
      MobArena: Disabled
      dungeon: Disabled
      world_nether: Survival
      world_the_end: Survival
    PreventGlobalMonsterEggs: true
    PreventTheft: true
    ProtectCreatures: true
    PreventButtonsSwitches: true
    LockWoodenDoors: false
    LockTrapDoors: false
    LockFenceGates: true
    EnderPearlsRequireAccessTrust: true
    RaidTriggersRequireBuildTrust: true
    ProtectHorses: true
    ProtectDonkeys: true
    ProtectLlamas: true
    InitialBlocks: 500
    Claim Blocks Accrued Per Hour:
      Default: 100
    Max Accrued Claim Blocks:
      Default: 25000
    Accrued Idle Threshold: 0
    AccruedIdlePercent: 0
    AbandonReturnRatio: 1.0
    AutomaticNewPlayerClaimsRadius: 4
    AutomaticNewPlayerClaimsRadiusMinimum: 0
    ExtendIntoGroundDistance: 5
    MinimumWidth: 5
    MinimumArea: 100
    MaximumDepth: -2147483648
    InvestigationTool: STICK
    ModificationTool: GOLDEN_SHOVEL
    Expiration:
      ChestClaimDays: 15
      UnusedClaimDays: 30
      AllClaims:
        DaysInactive: 35
        ExceptWhenOwnerHasTotalClaimBlocks: 25000
        ExceptWhenOwnerHasBonusClaimBlocks: 5000
      AutomaticNatureRestoration:
        SurvivalWorlds: false
    AllowTrappedInAdminClaims: false
    MaximumNumberOfClaimsPerPlayer: 30
    CreationRequiresWorldGuardBuildPermission: true
    VillagerTradingRequiresPermission: true
    CommandsRequiringAccessTrust: /sethome
    DeliverManuals: true
    ManualDeliveryDelaySeconds: 30
    RavagersBreakBlocks: true
    FireSpreadsInClaims: false
    FireDamagesInClaims: false
    LecternReadingRequiresAccessTrust: true
  Spam:
    Enabled: false
    LoginCooldownSeconds: 60
    LoginLogoutNotificationsPerMinute: 5
    ChatSlashCommands: /me;/global;/local
    WhisperSlashCommands: /tell;/pm;/r;/whisper;/msg
    WarningMessage: Please reduce your noise level.  Spammers will be banned.
    BanOffenders: true
    BanMessage: Banned for spam.
    AllowedIpAddresses: 1.2.3.4; 5.6.7.8
    DeathMessageCooldownSeconds: 120
    Logout Message Delay In Seconds: 0
  PvP:
    RulesEnabledInWorld:
      world: true
      world_nether: true
      world_the_end: true
      clanwars: true
      QueenBeeWorld: true
      arena: true
      dungeon: true
      Boss: true
      spawn: true
      MobArena: true
      resource_world: true
    ProtectFreshSpawns: true
    PunishLogout: true
    CombatTimeoutSeconds: 15
    AllowCombatItemDrop: false
    BlockedSlashCommands: /home;/vanish;/spawn;/tpa
    ProtectPlayersInLandClaims:
      PlayerOwnedClaims: true
      AdministrativeClaims: true
      AdministrativeSubdivisions: true
    AllowLavaDumpingNearOtherPlayers:
      PvPWorlds: true
      NonPvPWorlds: false
    AllowFlintAndSteelNearOtherPlayers:
      PvPWorlds: true
      NonPvPWorlds: false
    ProtectPetsOutsideLandClaims: false
  Economy:
    ClaimBlocksMaxBonus: 5000
    ClaimBlocksPurchaseCost: 0.0
    ClaimBlocksSellValue: 0.0
  ProtectItemsDroppedOnDeath:
    PvPWorlds: false
    NonPvPWorlds: true
  BlockLandClaimExplosions: true
  BlockSurfaceCreeperExplosions: true
  BlockSurfaceOtherExplosions: true
  LimitSkyTrees: true
  LimitTreeGrowth: false
  PistonMovement: CLAIMS_ONLY
  PistonExplosionSound: true
  FireSpreads: false
  FireDestroys: false
  AdminsGetWhispers: true
  AdminsGetSignNotifications: true
  VisualizationAntiCheatCompatMode: false
  SmartBan: true
  Mute New Players Using Banned Words: true
  MaxPlayersPerIpAddress: 3
  SilenceBans: true
  Siege:
    Worlds: []
    BreakableBlocks:
    - GRASS_BLOCK
    - DIRT
    - COBBLESTONE
    - OAK_PLANKS
    - SPRUCE_PLANKS
    - BIRCH_PLANKS
    - JUNGLE_PLANKS
    - ACACIA_PLANKS
    - DARK_OAK_PLANKS
    - SAND
    - GRAVEL
    - GLASS
    - GRASS     <-- can't understand this entry, see BukkitDev documentation
    - FERN
    - DEAD_BUSH
    - WHITE_WOOL
    - ORANGE_WOOL
    - MAGENTA_WOOL
    - LIGHT_BLUE_WOOL
    - YELLOW_WOOL
    - LIME_WOOL
    - PINK_WOOL
    - GRAY_WOOL
    - LIGHT_GRAY_WOOL
    - CYAN_WOOL
    - PURPLE_WOOL
    - BLUE_WOOL
    - BROWN_WOOL
    - GREEN_WOOL
    - RED_WOOL
    - BLACK_WOOL
    - SNOW
    - GLASS_PANE
    DoorsOpenDelayInSeconds: 300
    CooldownEndInMinutes: 60
  EndermenMoveBlocks: false
  SilverfishBreakBlocks: false
  CreaturesTrampleCrops: false
  RabbitsEatCrops: true
  HardModeZombiesBreakDoors: false
  MobProjectilesChangeBlocks: false
  Database:
    URL: 
    UserName: 
    Password: 
  UseBanCommand: false
  BanCommandPattern: ban %name% %reason%
  Advanced:
    fixNegativeClaimblockAmounts: true
    ClaimExpirationCheckRate: 60
    OfflinePlayer_cache_days: 90
  Abridged Logs:
    Days To Keep: 7
    Included Entry Types:
      Social Activity: true
      Suspicious Activity: true
      Administrative Activity: false
      Debug: false
      Muted Chat Messages: false
  ConfigVersion: 1
```

### Running without GriefPrevention

- [ ] I attempted running the server without GriefPrevention installed.
- [ ] The problem does _not_ occur when GriefPrevention is removed from the server.

### Running with only GriefPrevention

- [ ] I attempted running only GriefPrevention on the server.
- [ ] The issue still occurs when GriefPrevention is the only plugin running.

### Running on a fresh, clean server installation

- [ ] I attempted testing for the issue on a new server.
- [ ] The issue still occurs on a new server.

### Using unmodified client

- [ ] I attempted testing for the issue with the vanilla client.
- [ ] The issue still occurs when using the vanilla client.

### We appreciate you taking the time to fill out a bug report!

- [X] I searched for similar issues before submitting this bug report.

**Comment by bigpresh** [comment]
Sounds like another report of #666 (and similar tickets closed in favour of #666 - e.g. #979, #1599, #2189 ...)

It only seems to happen when using the MySQL data storage rather than flat files (unless your server ran out of disk space...)

Can you confirm if you're using MySQL data storage and just removed the config for it from your config.yml paste above for security, or whether you're using flat files - and also whether your server could have run out of disk space when the problem occurred?

**Comment by sennby** [comment]
> Sounds like another report of #666 (and similar tickets closed in favour of #666 - e.g. #979, #1599, #2189 ...)
> 
> It only seems to happen when using the MySQL data storage rather than flat files (unless your server ran out of disk space...)
> 
> Can you confirm if you're using MySQL data storage and just removed the config for it from your config.yml paste above for security, or whether you're using flat files - and also whether your server could have run out of disk space when the problem occurred?

correct

**Comment by Jikoo** [comment]
As stated in the other issues bigpresh linked, GP's SQL implementation should be avoided unless absolutely necessary. It is strictly worse than flatfile unless you need to read data (but not write! GP will clobber edits to loaded data!) from another system that supports SQL. If you do need to modify GP's data, that can only fully reliably be done via commands.
Closing in favor of #666.
